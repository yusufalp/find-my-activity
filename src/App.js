import React from 'react';
import { Route } from 'react-router-dom';
import Footer from './Footer/Footer';
import Navbar from './Navbar/Navbar';
import Home from './Home/Home';
import FindActivity from './FindActivity/FindActivity';
import AddActivity from './AddActivity/AddActivity';
import About from './About/About';
import Contact from './Contact/Contact';
import ActivityContext from './ActivityContext/ActivityContext';
import config from './config';

class App extends React.Component {
  state = {
    categories: [],
    activities: []
  }
  componentDidMount() {
    this.getData();
  }
  getData = () => {
    const url = config.API_ENDPOINT
    fetch(`${url}/api/categories`)
      .then(res => res.json())
      .then(categories => {
        this.setState({
          categories
        })
        return fetch(`${url}/api/activities`)
      })
      .then(res => res.json())
      .then(activities => {
        this.setState({
          activities
        })
      })
      .catch(err => console.log(err))
  }
  render() {
    const contextValue = {
      categories: this.state.categories,
      activities: this.state.activities,
      getData: this.getData
    }
    return (
      <div className="App">
        <ActivityContext.Provider value={contextValue} >
          <div className="mainpage-header">
          </div>
          <Navbar />
          <main>
            <Route
              exact path='/'
              component={Home}
            />
            <Route
              path='/find'
              component={FindActivity}
            />
            <Route
              path='/add'
              component={AddActivity}
            />
            <Route
              path='/about'
              component={About}
            />
            <Route
              path='/contact'
              component={Contact}
            />
          </main>
          <Footer />
        </ActivityContext.Provider>
      </div>
    );
  }
}

export default App;
