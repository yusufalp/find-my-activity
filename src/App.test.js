import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom'
import App from './App';
import renderer from 'react-test-renderer';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <App />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <App />
        </MemoryRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})