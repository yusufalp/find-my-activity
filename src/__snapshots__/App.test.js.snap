// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<div
  className="App"
>
  <div
    className="mainpage-header"
  />
  <nav
    className="navigation-bar"
    id="top-navbar"
  >
    <a
      className="brand-logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Brand logo"
        src="favicon.ico"
      />
    </a>
    <p
      className="menu-toggler"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
      />
    </p>
    <div
      className="navbar-content open-menu"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="navbar-link"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="navbar-link"
        >
          <a
            href="/find"
            onClick={[Function]}
          >
            Find Activity
          </a>
        </li>
        <li
          className="navbar-link"
        >
          <a
            href="/add"
            onClick={[Function]}
          >
            Add Activity
          </a>
        </li>
        <li
          className="navbar-link"
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="navbar-link"
        >
          <a
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <main>
    <div>
      <header
        className="home-header"
      >
        <h1>
          Find My Activity
        </h1>
        <a
          className="home-links"
          href="/find"
          onClick={[Function]}
        >
          Find Activity
        </a>
        <a
          className="home-links"
          href="/add"
          onClick={[Function]}
        >
          Add Activity
        </a>
      </header>
      <div
        className="group"
      >
        <div
          className="item"
        >
          <img
            alt="timer"
            className="feature-img"
            src="/images/chronometer.png"
          />
          <h3>
            Save time
          </h3>
          <p>
            Sure you can search for many activities online however, Find My Activity is the only place you can find it right away, We do not let you look for the needle in the hay
          </p>
        </div>
        <div
          className="item"
        >
          <img
            alt="group of people"
            className="feature-img"
            src="/images/group.png"
          />
          <h3>
            Give back to community
          </h3>
          <p>
            You can share your favorite activities with our community and we are forever grateful for that. After all, it takes a village, am I right?
          </p>
        </div>
        <div
          className="item"
        >
          <img
            alt="happy family"
            className="feature-img"
            src="/images/happiness.png"
          />
          <h3>
            Enjoy
          </h3>
          <p>
            These activities are meant for kids (and kids inside you). So you can either enjoy it by yourself or the whole family
          </p>
        </div>
      </div>
      <img
        alt="group of kids"
        className="group-kids"
        src="/images/group-kids.png"
      />
      <hr />
      <section>
        <h2>
          How to use this website?
        </h2>
        <p>
          There are two sections in this website where you can interact about the activities: Find and Add Activity.
        </p>
        <p>
          In the Find Activity section, search activities by the category. Choose outdoors if you want your child to work on something outdoor. Do you want something more like academics related, we got that. How about sports? We are constantly increasing the amount of the categories as we see more requests.
        </p>
        <p>
          In the Add Activity section, you can add your own activity. Have you been doing an activity with your child, share it with our community. Simply add the details of the activity requested and it will be posted for our users to search for.
        </p>
        <p>
          Enjoy!
        </p>
      </section>
    </div>
  </main>
  <footer>
    <section>
      <h6>
        Copyrights Â© Yusuf Alp 
        2021
      </h6>
      <a
        className="social"
        href="https://www.linkedin.com/in/yusuf-alp/"
      >
        Linkedin
      </a>
      <a
        className="social"
        href="https://github.com/yusufalp"
      >
        Github
      </a>
      <a
        className="social"
        href="mailto:alpyusuf007@gmail.com"
      >
        Email
      </a>
    </section>
    <div
      id="attribute"
    >
      Icons made by 
      <a
        href="http://www.freepik.com/"
        title="Freepik"
      >
        Freepik
      </a>
       from 
      <a
        href="https://www.flaticon.com/"
        title="Flaticon"
      >
        www.flaticon.com
      </a>
    </div>
  </footer>
</div>
`;
